### PR TITLE
Feature/download cart #47

### LINKED ISSUE #47: [datagateway-download] Create "cart" view
This is a large issue that may get split up into smaller issues.

Recreate the TopCAT download cart. Unlike TopCAT which has it as a modal, we should instead have it as a separate page accessed via the menu (`Data/Download`). I think we shall need two "tabs" - one for the download cart and one for the download status page.

Cart mockup (based on current TopCAT):

cart

When the user clicks download, they should then be taken to the download confirmation page

confirm download

Acceptance criteria:
- [ ] Have a download cart that displays the datafiles/datasets the user has selected
- [ ] The user can remove individual cart items or empty the whole cart
- [ ] When the user clicks download cart they are taken to the confirm download page
- [ ] The cart displays info on the total number of files and size of files in the cart, with maximums for Diamond

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Adds the download cart view that shows the user the items they currently have in their cart and gives them the opportunity to remove items or to proceed and download their cart.

Currently, the URLs required (topcat and datagateway-api) are not configurable since there is no settings file yet - instead this is being tracked in a separate issue: #88

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [x] Review code
- [x] Check Travis build
- [x] Review changes to test coverage

## Agile board tracking
Closes #47

### CODE REVIEW

**Comment by esl-lewis** [comment]
Some minor UI suggestions, I expected more feedback from the remove dataset button. Maybe it could flash a different colour when pressed.

**Comment by esl-lewis** [comment]
I ran the general npm tests for both dg-download and dg-common, these passed fine. But when I ran cypress interactively a few tests failed- more specifically when trying to access a row by aria label in 'should load correctly and display cart items' and 'should be able to sort cart items by name and type'. 
I assume this is due to the authentication issue we discussed earlier and 'npm test' works fine so I have approved anyway.

File: packages/datagateway-download/src/downloadCart/downloadCartApi.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        items: `${entityType} ${entityId}`,
+      },
+    })
+    .then(() => {})
+    .catch(error => {
+      log.error(error.message);
+    });
+};
+
+// TODO: caching???
```
Comment (esl-lewis) [inline packages/datagateway-download/src/downloadCart/downloadCartApi.ts]: Do you want this in with this commit?

**Comment by louise-davies** [comment]
@esl-lewis 

- sort by Size
This could probably be implemented, the only reason it wasn't is 1. because TopCAT doesn't and 2.  because this value is dynamically loaded and so there would have to be some handling of disabling until everything has attempted to load / dealing with failed cases. I'll see how simple it is to add.

- details panels
This one is more tricky - we aren't actually loading the information on the entities here - the TopCAT cart stores the entity type and entity name, and we query for the size, but any other information would also have to be queried. This could be done, but I would rather wait for user feedback/make it a separate feature request before performing work for a function that might not be wanted by users.

- UI feedback upon item removal
I'll have a go at delaying the removal for a bit so that I can give more visual feedback

**Comment by louise-davies** [comment]
@esl-lewis 
Here's a demo of adding a small delay and changing the colour of the icon when the item is being removed - does this feel better?

remove-cart-color-change

File: packages/datagateway-download/src/downloadCart/downloadCartApi.ts (hunk 1 above)
Comment (louise-davies) [reply packages/datagateway-download/src/downloadCart/downloadCartApi.ts]: I've removed this comment, we think caching in TopCAT was a solution for when getSize calls were expensive - so we'll only look into caching if we observe performance issues.

**Comment by louise-davies** [comment]
@esl-lewis I've implemented the changes you suggested in your last review - could you please review this again?

**Comment by louise-davies** [comment]
@esl-lewis This uses the `formatBytes` method, which currently always rounds to 2 decimal places. There's a trade-off of accuracy and information overload. For that specific figure, rounding is a concern since it is presenting a limit to the user and so displaying inaccurate information may cause issues in that the UI may show as being less than the limit but the actual number is greater than. It's not a difficult issue to fix if we get user feedback that we need variability on how many decimals/significant figures we should round to.

**Comment by esl-lewis** [comment]
@louise-davies fair enough, I guess you wouldn't want to display inaccurate information.

**Pull request merged**